stake-pool/js/src/index.ts

```diff
@@ -243,37 +166,33 @@ export async function depositSol(
     }),
   );
 
-  const { poolMint } = stakePool;
-
   // Create token account if not specified
-  if (!poolTokenReceiverAccount) {
-    poolTokenReceiverAccount = await addAssociatedTokenAccount(
+  if (!destinationTokenAccount) {
+    destinationTokenAccount = await addAssociatedTokenAccount(
       connection,
       from,
-      poolMint,
+      stakePool.poolMint,
       instructions,
     );
   }
 
-  const depositAuthority = undefined;
-
   const withdrawAuthority = await findWithdrawAuthorityProgramAddress(
-    StakePoolProgram.programId,
+    STAKE_POOL_PROGRAM_ID,
     stakePoolAddress,
   );
 
   instructions.push(
-    StakePoolProgram.depositSolInstruction({
-      stakePoolPubkey: stakePoolAddress,
-      depositAuthority,
-      withdrawAuthority,
-      reserveStakeAccount: stakePool.reserveStake,
-      lamportsFrom: userSolTransfer.publicKey,
-      poolTokensTo: poolTokenReceiverAccount,
+    StakePoolInstruction.depositSol({
+      stakePool: stakePoolAddress,
+      reserveStake: stakePool.reserveStake,
+      fundingAccount: userSolTransfer.publicKey,
+      destinationPoolAccount: destinationTokenAccount,
       managerFeeAccount: stakePool.managerFeeAccount,
-      referrerPoolTokensAccount: referrerTokenAccount ?? poolTokenReceiverAccount,
-      poolMint,
+      referralPoolAccount: referrerTokenAccount ?? destinationTokenAccount,
+      poolMint: stakePool.poolMint,
       lamports,
+      withdrawAuthority,
+      depositAuthority: undefined,
```

Comment:
So this still won't be passed in? not a huge a deal, but with a parameter that defaults to `undefined`, this becomes instantly usable!